python/ray/tune/logger.py

```diff
@@ -255,10 +257,18 @@ def default(self, value):
         try:
             if np.isnan(value):
                 return None
-            if np.issubdtype(value, float):
-                return float(value)
-            if np.issubdtype(value, int):
+
+            if (type(value).__module__ == np.__name__
+                    and isinstance(value, np.ndarray)):
+                return value.tolist()
+
+            if issubclass(type(value), numbers.Integral):
                 return int(value)
+            if issubclass(type(value), numbers.Number):
+                return float(value)
+
+            return super(_SafeFallbackEncoder, self).default(value)
```

Comment:
This looks ok, but is it necessary?

Reply:
I don't know if it's necessary but it seemed weird that there could potentially be values that just go past this try block without being properly encoded (i.e. encoded as `None`).